cpp/src/io/parquet/reader_impl.cu

```diff
@@ -278,15 +278,27 @@ struct metadata : public FileMetaData {
    *
    * @param row_group Index of the row group to select
    * @param max_rowgroup_count Max number of consecutive row groups if > 0
+   * @param row_group_indices Arbitrary rowgroup list[max_rowgroup_count] if non-null
    * @param row_start Starting row of the selection
    * @param row_count Total number of rows selected
    *
    * @return List of row group indexes and its starting row
    */
-  auto select_row_groups(int row_group, int max_rowgroup_count, int &row_start, int &row_count) {
+  auto select_row_groups(int row_group, int max_rowgroup_count,
+                         const size_type *row_group_indices,
+                         int &row_start, int &row_count) {
     std::vector<std::pair<int, int>> selection;
 
-    if (row_group != -1) {
+    if (row_group_indices) {
+      row_count = 0;
+      for (int i = 0; i < max_rowgroup_count; i++) {
+        auto rowgroup_idx = row_group_indices[i];
+        CUDF_EXPECTS(rowgroup_idx >= 0 && rowgroup_idx < get_num_row_groups(), "Invalid rowgroup index");
```

Comment:
Should have a test that this error is thrown as expected.

Reply:
Seems like low !/$ (like negative), but okay (I would only have been surprised if that suggestion came from anybody else :) )

Reply:
Do explain? Generally we should test that our code correctly handles exceptional situations, corner cases, and malformed inputs.

Reply:
A dozen lines of test code (estimate) to test for the presence of a 1-line CUDF_EXPECT() whose purpose is to indicate a bug in the calling code. (the very existence of that line implies the check is there, so a test seems redundant to me ~ somebody will have to maintain that, it's not free, hence !/$ should be a consideration)
On the other hand, I just typed more characters than what would be needed to add the test, so there's that.

Reply:
@OlivierNV to push back a bit, Cython will catch and raise these errors as Python exceptions which users can then catch and change their logical flow based on which is often the "Pythonic Way" of doing things.

I.E. someone may have some type of flow where they ask to read only the 1st rowgroup of a Parquet file for some reason and would catch this error to handle the 0 rowgroup case.

We generally try to test that we're properly throwing exceptions on the Python side as well.